### PR TITLE
Added interface

### PR DESCRIPTION
**Pull request by jezekr**
This is an interface for Octave

### CODE REVIEW

File: bin/runcutest [hunk 1]
```diff
@@ -467,7 +470,7 @@ if [[ ${PKG} == "matlab" ]]; then
         error "${thisprog}: environment variable MYMATLAB not set"
         exit 10
     fi
-    MEXFORTRAN=${MYMATLAB}/bin/mex
+    MEXFORTRAN=${MYMATLAB}/bin/MEX
```
Comment (nimgould) [inline bin/runcutest:473]: Why has this changed from mex to MEX? There is no binary MEX with my Linux Matlab, only mex. Is this an issue with Matlab on non-Linux systems?

**Review by nimgould — COMMENTED** [review]
My only comment is about mx <-> MEX in runcutest, but we need to agree before we merge

File: bin/runcutest (hunk 1 above)
Comment (jezekr) [reply bin/runcutest:473]: Oh, I think I forgot to remove this.

**Comment by dpo** [comment]
Many thanks @jezekr ! Because Octave is free, it would be great to add test for your interface in the continuous integration scripts. This way, we can be sure it continues to work as intended over time.

**Review by nimgould — APPROVED** [review]
That is super, thank you. I will now merge

**Comment by nimgould** [comment]
I am a bit concerned that when I complie here, I get

 Compiling mcutest        		gcc -c -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/include/octave-6.4.0/octave/.. -I/usr/include/octave-6.4.0/octave   -fopenmp -g -O2 -ffile-prefix-map=/build/octave-LVK0E0/octave-6.4.0=. -flto=auto -ffat-lto-objects -flto=auto -ffat-lto-objects -fstack-protector-strong -Wformat -Werror=format-security -g   -I. -I/usr/include/octave-6.4.0/octave/ -I/home/nimg/fortran/optrove_git/cutest/include  -DMEX_DEBUG mcutest.c -o mcutest.o
mcutest.c: In function ‘SparseVector’:
mcutest.c:1982:8: warning: assignment to ‘int *’ from incompatible pointer type ‘mwIndex *’ {aka ‘long int *’} [-Wincompatible-pointer-types]
 1982 |     ir = mxGetIr(vector);
      |        ^
mcutest.c:1983:10: warning: assignment to ‘int *’ from incompatible pointer type ‘mwIndex *’ {aka ‘long int *’} [-Wincompatible-pointer-types]
 1983 |     jptr = mxGetJc(vector);
      |          ^
mcutest.c: In function ‘extractSparseVector’:
mcutest.c:2019:8: warning: assignment to ‘int *’ from incompatible pointer type ‘mwIndex *’ {aka ‘long int *’} [-Wincompatible-pointer-types]
 2019 |     ir = mxGetIr(vector);
      |        ^
mcutest.c:2020:10: warning: assignment to ‘int *’ from incompatible pointer type ‘mwIndex *’ {aka ‘long int *’} [-Wincompatible-pointer-types]
 2020 |     jptr = mxGetJc(vector);
      |          ^
mcutest.c: In function ‘coordToMatlabSparse’:
mcutest.c:2066:10: warning: assignment to ‘int *’ from incompatible pointer type ‘mwIndex *’ {aka ‘long int *’} [-Wincompatible-pointer-types]
 2066 |     ir   = mxGetIr(matrix);  /* Array of length nnz    */
      |          ^
mcutest.c:2067:10: warning: assignment to ‘int *’ from incompatible pointer type ‘mwIndex *’ {aka ‘long int *’} [-Wincompatible-pointer-types]
 2067 |     jptr = mxGetJc(matrix);  /* Array of length ncol+1 */

Getting the correct tie between Mex types is always painful, particularly when Matlab kept changing its mind. It seems as if mxGetIr/mexJc in octave may not be long integers. Is this an issue ... it only seems to be a compiler warning

**Comment by jezekr** [comment]
What was your command?

**Comment by nimgould** [comment]
make -f $CUTEST/makefiles/pc64.lnx.gfo all
in the $CUTEST/src/octave directory
It looks as if mxGetIr/mexJc have a different type
in octave than for Matlab.

Once I start octave (and having created a mcutest.mex file),
I seem to have a further problem - 

> prob = cutest_setup()
error: /home/nimg/problems/sif/mcutest.mex: failed to load: /home/nimg/problems/sif/mcutest.mex: undefined symbol: cutest_udimsh_
error: called from
    cutest_setup at line 10 column 20

This may be related to the earlier issue?

**Comment by jezekr** [comment]
mwIndex is defined in usr/include/$OCTAVE_VERSION/octave/mxarray.h

in my version I found:
/*
 * FIXME: Mathworks says mwSize, mwIndex should be int generally.
 * But on 64-bit systems, or when mex -largeArrayDims is used, it is size_t.
 * mwSignedIndex is supposed to be ptrdiff_t.  All of this is confusing.
 * Its better to conform to the same indexing as the rest of Octave.
 */
typedef int mwIndex;

but in every version of Octave it is different.

**Comment by jezekr** [comment]
Have you enabled MATLAB functionalities when you installed CUTEST?

**Comment by nimgould** [comment]
I have the same comment, but in octave-6.4.0 the file sets

typedef int64_t mwSize;
typedef int64_t mwIndex;
typedef int64_t mwSignedIndex;

**Comment by nimgould** [comment]
Yes, everything is ok with Matlab.

**Comment by jezekr** [comment]
You should not enable MATLAB functionalities. Please install CUTEST without enabling MATLAB functionalities.

Before you setup a problem always clear mcutest. This is necessary in Octave. Maybe I should add this to the documentary.

**Comment by nimgould** [comment]
I am not sure I understand, octave generates mcutest.mex, while matlab generates mcutest.mexa64. What else should I remove (yes, this needs to be in the README)?

**Pull request merged**